### DISCUSSION TITLE
I wrote a quick guide for adding actually working custom portals with password verification using the Custom-Portals plugin

### DISCUSSION THREAD

**Discussion by minanagehsalalma** [KeyofBlueS/airgeddon-plugins#19] [General]
### Add the guide to it if you want

1. Find **`'`** (a singel quote) and replace with **`'"'"'`** in the whole html fie 

2. Enable Regular expression Find `^` and replace with `echo -e '`

3.  Find `$` and replace with **`'`** 

4.  add these lines to the start of the file 

```
#!/usr/bin/env bash
echo '<!DOCTYPE html>'
echo '<html>'
```

5. To get the password verification to work , The input bar form which the value you want is in has to have a form tag with action url set to check.htm just like this 

```
<form method="post" id="loginform" name="loginform" action="check.htm">
<input id="password" type="password" name="password" maxlength="63" size="20" />
</form>

```
6. Use the custom_portals plugin by placing the .sh file in airgeddon plugins folder and then place the webpage files index.htm ..etc in a new folder with any name you like in the custom_portals folder 
and if you want to edit the check.htm file then you have to run the script with generic settings then get the files from path 
tmp/www and after editing it add it next to your custom index.htm 

MAKE SURE THAT THE FILES ARE  .htm not .HTML
The link to download the plugin 

https://github.com/KeyofBlueS/airgeddon-plugins

7. start airgeddon script and before being presented to choose the language of the captive portal 
you will get to choose the captive portal page from the custom_portals folder 

Thanks for xpz3 & OscarAkaElvis for clearing things Up ❤

Thanks to you too : )

**Comment by xpz3** [comment]
Step 1 should have been at step 5 i think @minanagehsalalma

**Comment by minanagehsalalma** [comment]
> Step 1 should have been at step 5 i think @minanagehsalalma

Done

**Comment by xpz3** [comment]
Using this you can port fluxion portals for use with airgeddon

**Comment by minanagehsalalma** [comment]
> Using this you can port fluxion portals for use with airgeddon

Sure, might make a repo containing them haha 

but you need to get hold of them first 
they aren't just lying around , you have to extract them from tmp/fluxion after running the script

**Comment by xpz3** [comment]
> > Using this you can port fluxion portals for use with airgeddon
> 
> Sure, might make a repo containing them haha
> 
> but you need to get hold of them first they aren't just lying around , you have to extract them from tmp/fluxion after running the script

No they're already in fluxion\attacks\Captive Portal
There are two types of portals. You'll know when you port

**Comment by minanagehsalalma** [comment]
@xpz3 

> No they're already in fluxion\attacks\Captive Portal
> There are two types of portals. You'll know when you port

Yeah , i am talking about the Generic ones

**Comment by koko9097** [comment]
mina pelase ca i have ur discord pelase habibi i realy need ur help

**Comment by xpz3** [comment]
> @xpz3 i didn't reply for a reason, delete your comment pls.

Done

**Comment by minanagehsalalma** [comment]
@KeyofBlueS since the count of people that know about issues are way more than whom know about discussions, i suggest that you mention the link here at the readme 

![image](https://user-images.githubusercontent.com/20546638/226097371-bf66b2a3-84e3-43df-b601-1d8067230e34.png)

**Comment by Cippo2019** [comment]
Hi and thanks for your work
I am trying to understand if we can use here the captive portal (like Dlink one for istance) with that customization

have any of you done this ?
thanks in advance.

Tz

**Comment by xpz3** [comment]
> ### Add the guide to it if you want
> 1. Find **`'`** (a singel quote) and replace with **`'"'"'`** in the whole html fie
> 2. Enable Regular expression Find `^` and replace with `echo -e '`
> 3. Find `$` and replace with **`'`**
> 4. add these lines to the start of the file
> 
> ```
> #!/usr/bin/env bash
> echo '<!DOCTYPE html>'
> echo '<html>'
> ```
> 
> 5. To get the password verification to work , The input bar form which the value you want is in has to have a form tag with action url set to check.htm just like this
> 
> ```
> <form method="post" id="loginform" name="loginform" action="check.htm">
> <input id="password" type="password" name="password" maxlength="63" size="20" />
> </form>
> ```
> 
> 6. Use the custom_portals plugin by placing the .sh file in airgeddon plugins folder and then place the webpage files index.htm ..etc in a new folder with any name you like in the custom_portals folder
>    and if you want to edit the check.htm file then you have to run the script with generic settings then get the files from path
>    tmp/www and after editing it add it next to your custom index.htm
> 
> MAKE SURE THAT THE FILES ARE .htm not .HTML The link to download the plugin
> 
> https://github.com/KeyofBlueS/airgeddon-plugins
> 
> 7. start airgeddon script and before being presented to choose the language of the captive portal
>    you will get to choose the captive portal page from the custom_portals folder
> 
> Thanks for xpz3 & OscarAkaElvis for clearing things Up ❤
> 
> Thanks to you too : )

I've made a new plugin that doesn't require any of these. Check it out in my git or airgeddon discord.
